### PR TITLE
#174797518 Login using email and password

### PR DESCRIPTION
**Pull request by Habinezajanvier**
#### What does this PR do?
Enable signin using email and password
#### Description of Task to be completed?
- Checking for email password matching
- Providing the user with JW token 
#### How should this be manually tested?
- Clone the repo `git clone https://github.com/atlp-rwanda/script-heroes-bn-backend.git`
- Switch your working `cd script-heroes-bn-backend`
- Checkout the branch `git checkout ft-email-login-#174797518`
- Create a `.env` file in the root folder and update it using data found in `env-example`
- Install all dependencies `npm install`
- Run migrations `sequelize db:migrate`
- Start the server `npm run dev`
- Using postman on Signup endpoint create an account on the app
    - Endpoint for signup ```http://localhost:3000/api/auth/signup``` 
- Still on postman Login using Email and password used to create the account
    - Endpoint for login ```http://localhost:3000/api/auth/login```
> You should see successful login message and the returned token
#### Any background context you want to provide?
- Number 3000 in endpoints may depend on the port you put in your `.env` 
- Data for creating an account should look like:
```json
{
    "firstName": "firstName",
    "lastName": "lastName",
    "email": "email@test.com",
    "phoneNumber": "0780000000",
    "password": "password"
}
```
- Data for login should look like
```json
{
    "email": "email@test.com",
    "password": "password"
}
```
- Response after login should look like
```json
{
    "message": "Your are successfully loged in",
    "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MTksImZpcnN0TmFtZSI6IkhhYmluZXphIiwibGFzdE5hbWUiOiJqYW52aWVyIiwiZW1haWwiOiJoYWJpbmV6YWphbnZAZ21haWwuY29tIiwicGhvbmVOdW1iZXIiOiIwNzg5MTIxMzI0IiwiaWF0IjoxNjAxMzk5NTM5LCJleHAiOjE2MDE0ODU5Mzl9.sGPkCF17q8sl6AibrrFcdSYfdsjoVkJuMIxPeO5X0dE"
}
```
#### What are the relevant pivotal tracker stories?
[#174797518](https://www.pivotaltracker.com/story/show/174797518)
#### Screenshots if availabe
image

### CODE REVIEW

**Review by dushimeemma — APPROVED** [review]
Good job

File: README.md [hunk 1]
```diff
@@ -56,4 +56,5 @@ Note: If you make changes that uses environmental variables make sure to add tho
 
 - Send a post request using postman
 - Hosted url: https://script-heroes-bn-backend-stgng.herokuapp.com/
-- Local url: http://localhost:3000/api/signup
+- Local url for signup: http://localhost:3000/api/auth/signup
+- Local url for login: http://localhost:3000/api/auth/login
```
Comment (kipropbrian) [inline README.md]: No need for this URLS.

File: src/utils/internationalization/locales/en.json [hunk 2]
```diff
@@ -1,5 +1,34 @@
...
+	"Password must be string": "Password must be string",
+	"Please fill in your Password": "Please fill in your Password",
+	"Password must be atleast {#limit} characters long": "Password must be atleast {#limit} characters long",
+	"Password is required": "Password is required",
+	"email should be a valid email": "email should be a valid email",
+	"please fill the email input": "please fill the email input",
+	"Email field can't empty": "Email field can't empty",
+	"Email should look like test@email.com": "Email should look like test@email.com",
+	"Please fill in the email input": "Please fill in the email input"
+}
```
Comment (kipropbrian) [inline src/utils/internationalization/locales/en.json]: Please add a new line at the end

**Comment by kipropbrian** [comment]
Include postman screenshots

File: src/controllers/emailAuth.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      return res
+        .status(404)
+        .json({ message: res.__('Email or password is incorrect') });
+    }
+    const validPass = await bcrypt.compare(password, userAccount.password);
+    if (!validPass) {
+      return res
+        .status(400)
+        .send({ message: res.__('Email or password is incorrect') });
+    }
```
Comment (kipropbrian) [inline src/controllers/emailAuth.js]: You can merge these two errors into one.

File: package.json [hunk 4]
```diff
@@ -10,8 +10,12 @@
     "start": "npm run build && node dist/index",
     "dev": "nodemon --exec babel-node src/index.js",
     "build": ".\\node_modules\\.bin\\babel src --out-dir dist",
```
Comment (kipropbrian) [inline package.json]: remove the unnecessary path

**Review by elvisiraguha — APPROVED** [review]

**Review by valens10 — APPROVED** [review]

**Review by Adadie — APPROVED** [review]

**Review by Adadie — COMMENTED** [review]
Add paths for the endpoints in the readme file

**Pull request merged**